common/src/main/java/io/pravega/common/concurrent/Futures.java

```diff
@@ -44,6 +45,31 @@
  */
 public final class Futures {
 
+    /**
+     * Makes sure that the result future given out to caller is actually completed on caller's executor.
+     * So any chaining, if done without specifying an executor, will either happen on
+     * caller's executor or fork join pool but never on someone else's executor.
+     *
+     * @param future The future to execute.
+     * @param <T> The Type of the future's result.
+     * @param executor The executor for executing the future.
+     * @return <T>CompletableFuture<T> Result of execution, successful or exceptional.
+     */
+    public static <T> CompletableFuture<T> completeOn(CompletableFuture<T> future, final Executor executor) {
```

Comment:
This method does not have any unit tests. Please write some.

Reply:
This method refactored form of `withCompletion()` method in StreamMetadataStore and many methods in TableandStreamMetadataStores invoke this method. So it is well tested by other UTs though there are no direct UTs for this one...

Reply:
fixed.